spec.html

```diff
@@ -29107,14 +29115,9 @@ <h1>%StringIteratorPrototype%.next ( )</h1>
             1. If _position_ &ge; _len_, then
               1. Set _O_.[[IteratedString]] to *undefined*.
               1. Return CreateIterResultObject(*undefined*, *true*).
-            1. Let _first_ be the numeric value of the code unit at index _position_ within _s_.
-            1. If _first_ &lt; 0xD800 or _first_ &gt; 0xDBFF or _position_ + 1 = _len_, let _resultString_ be the String value consisting of the single code unit _first_.
-            1. Else,
-              1. Let _second_ be the numeric value of the code unit at index _position_ + 1 within the String _s_.
-              1. If _second_ &lt; 0xDC00 or _second_ &gt; 0xDFFF, let _resultString_ be the String value consisting of the single code unit _first_.
-              1. Else, let _resultString_ be the string-concatenation of the code unit _first_ and the code unit _second_.
-            1. Let _resultSize_ be the number of code units in _resultString_.
-            1. Set _O_.[[StringIteratorNextIndex]] to _position_ + _resultSize_.
+            1. Let _cp_ be ! CodePointAt(_s_, _position_).
+            1. Set _O_.[[StringIteratorNextIndex]] to _position_ + _cp_.[[CodeUnitCount]].
+            1. Let _resultString_ be the String value of length _cp_.[[CodeUnitCount]] equal to the substring of _s_ consisting of the code units at indices _position_ (inclusive) through _O_.[[StringIteratorNextIndex]] (exclusive).
```

Comment:
```suggestion
            1. Let _resultString_ be the String value containing _cp_.[[CodeUnitCount]] consecutive code units from _s_ beginning with the code unit at index _position_.
```
Same effect but a bit simpler.

Also, you could then swap the steps back so that `_O_.[[StringIteratorNextIndex]]` is set last, which I think makes slightly more sense.

Reply:
Updated.